pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -9164,6 +9164,450 @@ spec:
 	}
 }
 
+func TestReconciler_PipelineTaskMatrixWithRetries(t *testing.T) {
+	names.TestingSeed()
+
+	task := parse.MustParseTask(t, `
+metadata:
+  name: mytask
+  namespace: foo
+spec:
+  params:
+    - name: platform
+    - name: browser
+  steps:
+    - name: echo
+      image: alpine
+      script: |
+        echo "$(params.platform) and $(params.browser)"
+        exit 1
+`)
+
+	cms := []*corev1.ConfigMap{withEmbeddedStatus(withEnabledAlphaAPIFields(newFeatureFlagsConfigMap()), config.MinimalEmbeddedStatus)}
+	cms = append(cms, withMaxMatrixCombinationsCount(newDefaultsConfigMap(), 10))
+
+	tests := []struct {
+		name                string
+		trs                 []*v1beta1.TaskRun
+		prs                 []*v1beta1.PipelineRun
+		expectedPipelineRun *v1beta1.PipelineRun
+		expectedTaskRuns    []*v1beta1.TaskRun
+	}{{
+		name: "matrixed pipelinetask with retries, where one taskrun has failed and another one is running",
+		trs: []*v1beta1.TaskRun{
+			mustParseTaskRunWithObjectMeta(t,
+				taskRunObjectMeta("pr-platforms-and-browsers-0", "foo",
+					"pr", "p", "platforms-and-browsers", false),
+				`
+spec:
+  params:
+  - name: platform
+    value: linux
+  - name: browser
+    value: chrome
+  resources: {}
+  serviceAccountName: test-sa
+  taskRef:
+    name: mytask
+  timeout: 1h0m0s
+  retries: 1
```

Comment:
why are there retries in the taskrun spec?

Reply:
our recent discussions about moving retries to be a taskrun concern put my mindset into having it as a taskrun concern, and guess the unknown fields for a given CRD are just ignored so it wasn't causing an issue, [removed](https://github.com/tektoncd/pipeline/compare/a9b466535a758375f3b538c8c0e793edc9074441..75a1c00101e98cc012f3ecef35fcc6dbcbcda4c7) the retries now -- thank you! 

(guess i really want the separation of concerns 😂)

Reply:
LOL, if you put it in the tests it'll eventually become a feature?
I think what confused me is that our yaml parser tests accept fields that aren't in the spec. Filed https://github.com/tektoncd/pipeline/issues/5318